timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/LocalPaxosComponentsTest.java

```diff
@@ -65,9 +65,13 @@ public void setUp() throws IOException {
     public void newClientCanBeCreated() {
         PaxosLearner learner = paxosComponents.learner(CLIENT);
         learner.learn(PAXOS_ROUND_ONE, PAXOS_VALUE);
-        assertThat(learner.getGreatestLearnedValue()).isNotNull();
-        assertThat(learner.getGreatestLearnedValue().getLeaderUUID()).isEqualTo(PAXOS_UUID);
-        assertThat(learner.getGreatestLearnedValue().getData()).isEqualTo(PAXOS_DATA);
+
+        assertThat(learner.getGreatestLearnedValue())
+                .isPresent()
```

Comment:
Call to `isPresent` can be removed, [`hasValueSatisfying`](https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AbstractOptionalAssert.html#hasValueSatisfying-org.assertj.core.api.Condition-) will also assert that the value is present.

Reply:
you're right! Will change this

Reply:
can we go further:

```java
Optional<PaxosValue> value = learner.getGreatestLearnedValue();
assertThat(value).map(PaxosValue::getLeaderUUID).contains(PAXOS_UUID);
assertThat(value).map(PaxosValue::getData).contains(PAXOS_DATA);
```

Reply:
note, `map` is a method on the `OptionalAssert`

Reply:
(yep, changed to `OptionalAssert`'s map)